### PR TITLE
Adds 'fail on error' option to the script

### PR DESCRIPTION
**Pull request by skyluc**
And fixed one problem.
I see one other failure, but it is linked to a docker bug ([docker#9665](https://github.com/docker/docker/issues/9665)), which might go away when upgrading to a newer kernel.

### CODE REVIEW

File: mesos-docker/run/run.sh [hunk 1]
```diff
@@ -89,9 +90,6 @@ function generate_application_conf_file {
   sed -i -- "s@replace_with_docker_host_ip@$host_ip@g" "$target_location/mit-application.conf"
   sed -i -- "s@replace_with_spark_executor_ui@$spark_tgz_file@g" "$target_location/mit-application.conf"
   
-  #remove any temp file generated
-  rm "$target_location/mit-application.conf--"
```
Comment (dragos) [inline mesos-docker/run/run.sh]: Why do you remove this line? Is it causing an error?

Comment (skyluc) [inline mesos-docker/run/run.sh]: Yes, it is attempting to remove a file which is never created.
I think it was use as a temporary file for the sed operations during development.

**Comment by skonto** [comment]
Sed on mac creates temp files to complete the replacement, so this rm statement is needed. I would just us rm -f so no message is printed.

**Comment by skonto** [comment]
If i want to make the script fail-fast i think its easy to do it : bash -e script.sh.

**Comment by dragos** [comment]
LGTM

**Comment by dragos** [comment]
sorry 'bout that, but this needs a rebase as well @skyluc :)

**Pull request merged**